lib/elixir/test/elixir/kernel/deprecated_test.exs

```diff
@@ -0,0 +1,38 @@
+Code.require_file("../test_helper.exs", __DIR__)
+
+defmodule Kernel.DeprecatedTest do
+  use ExUnit.Case
+
+  import PathHelpers
+
+  test "raises on invalid @deprecated" do
+    assert_raise ArgumentError, ~r"@deprecated expects a string with the reason", fn ->
+      defmodule InvalidDeprecated do
+        @deprecated 1.2
```

Comment:
One extra comment here, I would expect some people only using `@deprecated` without any argument and this was my first test case, but I think the warning is not very good/clear in the case:

```
warning: module attribute @deprecated in code block has no effect as it is never returned (remove the attribute or assign it to _ to avoid warnings)
  lib/elixir/test/elixir/kernel/deprecated_test.exs:11

warning: undefined module attribute @deprecated, please remove access to @deprecated or explicitly set it before access
  lib/elixir/test/elixir/kernel/deprecated_test.exs:11: Kernel.DeprecatedTest.InvalidDeprecated (module)
```

I don't if it makes sense to improve this message specifically, to something like:

```
warning: @deprecated expects a string with the reason for the deprecation, got: nil
```

Reply:
> I don't if it makes sense to improve this message specifically, to something like:

I don't think it makes sense because differently from other languages `@foobar` in Elixir is always a read operation, never a write one.

Reply:
Yeah, this is why I didn't change. But if you are not used to the language, you can spend some time trying to understand what is exactly that. And to be fair, it's written on the second warning, it's just that you maybe get confused with it saying that the attribute is undefined and them talking about access. For me, maybe it would more clear if the warning was something like that:
```
warning: you are trying to use @deprecated, but it was never explicitly set before. If you are trying to set the attribute, please define its value after the attribute. Otherwise, remove the access to it.
```

But, in this case specifically, I was thinking that I did something wrong with the reading of the attribute, so I was reading "undefined module attribute @deprecated" like a different error. Maybe the warning is clear enough..

Reply:
@ggcampinho we can definitely rewrite the warning although I still think we should start with "undefined module attribute @deprecated". The initial reason should be compact, so those familiar with the warning can know what happened without reading the whole message. Then we expand the message.

So maybe something like:

```
undefined module attribute @deprecated. If you are trying to set the attribute, a value must be explicitly given as argument, otherwise remove the use of @deprecated
```

Reply:
It's better for me, should I change here or in other PR?

Reply:
Let's do it in another PR please!